fcrepo-persistence-ocfl/src/test/java/org/fcrepo/persistence/ocfl/impl/CreateRDFSourcePersisterTest.java

```diff
@@ -67,18 +40,39 @@
 import org.mockito.Mock;
 import org.mockito.junit.MockitoJUnitRunner;
 
+import java.io.InputStream;
+import java.time.Instant;
+import java.util.stream.Stream;
+
+import static org.apache.jena.graph.NodeFactory.createLiteral;
+import static org.apache.jena.graph.NodeFactory.createURI;
+import static org.apache.jena.rdf.model.ModelFactory.createDefaultModel;
+import static org.fcrepo.kernel.api.RdfLexicon.RDF_SOURCE;
+import static org.fcrepo.kernel.api.operations.ResourceOperationType.CREATE;
+import static org.fcrepo.persistence.common.ResourceHeaderSerializationUtils.RESOURCE_HEADER_EXTENSION;
+import static org.fcrepo.persistence.common.ResourceHeaderSerializationUtils.deserializeHeaders;
+import static org.fcrepo.persistence.ocfl.impl.OCFLPersistentStorageUtils.getInternalFedoraDirectory;
+import static org.fcrepo.persistence.ocfl.impl.OCFLPersistentStorageUtils.getRDFFileExtension;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static org.mockito.Mockito.withSettings;
+
 /**
  * @author dbernstein
  * @since 6.0.0
  */
 @RunWith(MockitoJUnitRunner.class)
-public class RDFSourcePersisterTest {
+public class CreateRDFSourcePersisterTest {
```

Comment:
Maybe add a test for what happens if you try to create a resource that already exists? And a test to make sure you can't create something with the same id as the repo root

Reply:
I believe we'll be checking at a higher level to make sure that you can create a resource that  already exists. Also,  when the fedora is initialized it will use this code path to create a repo root if none exists.

Reply:
I guess its a bit edgy, but concurrent requests could attempt to create the same resource (simultaneous PUTs to the same resource being the most likely), unless there is going to be a locking mechanism further up. It can be nice to be explicit about expectations, but we can move ahead without the check if you feel confident it should be handled elsewhere.

I'd still recommend having a test to verify that you can't create an RDF or Non-RDF resource with the same ocfl id as the root since you have an explicit check for that https://github.com/fcrepo4/fcrepo4/blob/c0108e45a8632253a55555e519457c1d6ebdc1e3/fcrepo-persistence-ocfl/src/main/java/org/fcrepo/persistence/ocfl/impl/AbstractPersister.java#L135

Reply:
okay.  will do.